test/IIS.Tests/ResponseAbortTests.cs

```diff
@@ -0,0 +1,154 @@
+                    await connection.Receive(
+                        "5",
+                        "Abort");
+                    bodyReceived.SetResult(true);
+
+                    await connection.WaitForConnectionClose();
+                }
+            }
+        }
+
+        [ConditionalFact]
+        public async Task ReadsThrowAfterAbort()
+        {
+            Exception exception = null;
+
+            using (var testServer = await TestServer.Create(
+                async ctx => {
+                    ctx.Abort();
+                    try
+                    {
+                        var a = new byte[10];
+                        await ctx.Request.Body.ReadAsync(a);
+                    }
+                    catch (Exception e)
+                    {
+                        exception = e;
+                    }
+                }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await SendContentLength1Post(connection);
+                    await connection.WaitForConnectionClose();
+                }
+            }
+
+            Assert.IsType<ConnectionAbortedException>(exception);
+        }
+
+        [ConditionalFact]
+        public async Task WritesNoopAfterAbort()
+        {
+            Exception exception = null;
+
+            using (var testServer = await TestServer.Create(
+                async ctx => {
+                    ctx.Abort();
+                    try
+                    {
+                        await ctx.Response.Body.WriteAsync(new byte[10]);
+                    }
+                    catch (Exception e)
+                    {
+                        exception = e;
+                    }
+                }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await SendContentLength1Post(connection);
+                    await connection.WaitForConnectionClose();
+                }
+            }
+
+            Assert.Null(exception);
+        }
+
+        [ConditionalFact]
+        public async Task RequestAbortedIsTrippedAfterAbort()
+        {
+            bool tokenAborted = false;
+            using (var testServer = await TestServer.Create(
+                ctx => {
+                    ctx.Abort();
+                    tokenAborted = ctx.RequestAborted.IsCancellationRequested;
+                    return Task.CompletedTask;
+                }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await SendContentLength1Post(connection);
+                    await connection.WaitForConnectionClose();
+                }
+            }
+
+            Assert.True(tokenAborted);
+        }
+
+        private static async Task SendContentLength1Post(TestConnection connection)
```

Comment:
Doesn't this function already exist somewhere else?

Reply:
They are a bit different.